### DISCUSSION TITLE
When a TIME OUT occurs we do not clean the go build folders in /tmp

### DISCUSSION THREAD

**Discussion by alessio-perugini** [go-gremlins/gremlins#175] [Development]
I noticed that when we have a TIME OUT we do not clean the folder created by go test in /tmp dir.

You can easily reproduce the behavior running gremlin with multiple workers in a big pkg. I tested on ./internal/coverage.

Checking the /tmp we can see that are left some folders like `go-build19625282`.

Step I did:
1. cd /tmp && rm -rf go-build*
2. gremlin unleash ./internal/coverage
3. ls /tmp | grep go-build.

Moreover, each TIMED OUT corresponds to a unique folder. If we have 5 TIMED OUT we will have 5 corresponding go-build* folder left in /tmp. You can lazy checking this by doing:
```bash
./dist/bin/gremlins unleash ./internal/coverage | grep "TIMED OUT" | wc -l
ls /tmp | grep go-build | wc -l
```

I'm not sure if we can have complete control over this. I think the problem is that we cannot know upfront the directory name created by `go test`. When the timeout occurs we shut down the command and the folder won't be cleaned up.

## Solutions
1. take all the folder starting with `go-build*`, put them in a map and clear all the go-build folder not already there.
2. Remove all go-build* folder before quitting gremlin, without performing any check
3. Open to new ideas

**Comment by k3rn31** [comment]
I would say we have a bug here. If this happens with tests I assume it happens also during a run with timeouts.

**Reply by alessio-perugini** [reply]
I found a better solution. Setting up the `GOTMPDIR` env when we call the `go test` will create the build dirs under the directory specified in the env. By doing so we can always clean all the builds created by gremlins. I'm opening an issue and start working on a draft
